pkg/controllers/scheduler/scheduler.go

```diff
@@ -361,6 +361,39 @@ func (s *Scheduler) prepareToSchedule(
 ) {
 	logger := klog.FromContext(ctx)
 
+	// check migration config
+
+	existingMigrationConfigBytes, migrationConfigExists := fedObject.GetAnnotations()[common.MigrationConfigurationAnnotation]
+	_, customMigrationInfoExists := fedObject.GetAnnotations()[common.AppliedMigrationConfigurationAnnotation]
+	if !migrationConfigExists {
+		if customMigrationInfoExists {
+			delete(fedObject.GetAnnotations(), common.AppliedMigrationConfigurationAnnotation)
```

Comment:
This relies on the fact that `fedObject.GetAnnotations()` returns a shallow copy, which is not necessarily the case for all types. e.g. if `fedObject` is an unstructured, `GetAnnotations()` returns a deep copy. A safer way to update the annotation requires calling `SetAnnotations()`.

Reply:
I just discovered this and used the current writing method. SetAnnotation may need to add an additional temporary variable.